configuration/azure/appconfig/appconfig.go

```diff
@@ -295,15 +295,21 @@ func (r *ConfigurationStore) Subscribe(ctx context.Context, req *configuration.S
 }
 
 func (r *ConfigurationStore) doSubscribe(ctx context.Context, req *configuration.SubscribeRequest, handler configuration.UpdateHandler, sentinelKey string, id string) {
+	var etagVal *azcore.ETag
 	for {
 		// get sentinel key changes
-		_, err := r.Get(ctx, &configuration.GetRequest{
-			Keys:     []string{sentinelKey},
-			Metadata: req.Metadata,
-		})
+		resp, err := r.client.GetSetting(
+			context.TODO(),
+			sentinelKey,
+			&azappconfig.GetSettingOptions{
+				Label:         r.getLabelFromMetadata(req.Metadata),
+				OnlyIfChanged: etagVal,
+			},
+		)
 		if err != nil {
```

Comment:
Please add a line here to not log errors if `errors.Is(err, context.Canceled)` or when the component is shutting down, an error would be logged